src/toolkit/matl_sell_policy.cc

```diff
@@ -153,8 +173,15 @@ std::set<BidPortfolio<Material>::Ptr> MatlSellPolicy::GetMatlBids(
     for (rit = requests.begin(); rit != requests.end(); ++rit) {
       req = *rit;
       qty = std::min(req->target()->quantity(), limit);
-      nbids = excl ? static_cast<int>(std::floor(qty / quantize_)) : 1;
-      qty = excl ? quantize_ : qty;
+      package_fill = package_->GetFillMass(qty);
+      if (package_fill == 0) {
+        nbids = 0;
+      } else {
+        nbids = excl ? static_cast<int>(std::floor(qty / quantize_)) : static_cast<int>(std::floor(qty / package_fill));
+      }
+      double remaining_qty = fmod(qty, package_fill);
+
+      qty = excl ? quantize_ : package_fill;
       for (int i = 0; i < nbids; i++) {
         m = buf_->Pop();
         buf_->Push(m);
```

Comment:
This isn't your code, but shouldn't we do this before the loop?

...and can't we `Peek()` instead of `Pop()` and `Push()`?

Also, what if multiple items need to be popped to fulfill an order?  Let's discuss...